opensrp-app/src/main/java/org/smartregister/domain/jsonmapping/util/Tree.java

```diff
@@ -112,6 +112,23 @@ public void addNode(K id, String label, T node, K parentId) {
         return null;
     }
 
+    /**
+     * Delete nodes from location hierarchy
+     *
+     * @param id the id of the node to remove
+     */
+    public void deleteNode(K id) {
+        TreeNode<K, T> node = getNode(id);
+        removeNode(id);
+        parentChildren.remove(id);
+        LinkedHashSet<K> parent = parentChildren.get(node.getParent());
+        if (parent != null && parent.size() == 1) {
+            deleteNode(node.getParent());
+        } else {
+            parent.remove(id);
```

Comment:
This is so that we don't have orphaned nodes. In the location tree we usually get the assigned locations and build the higher nodes, so if a lower node is removed, then its local the the parent should be removed

Reply:
Cool